### PR TITLE
circuit-types: fixed-point: Refactor fixed point arithmetic over new circuits

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR refactors the fixed point arithmetic implementation over the new `mpc-jellyfish` circuits. This includes types for native FP arithmetic, in-circuit arithmetic, and mpc fixed point arithmetic.]

### Testing
- All unit tests pass
- Tested arithmetic operations of all forms of fixed point

### CODE REVIEW

File: circuit-macros/src/circuit_type/singleprover_circuit_types.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
 //! Groups type and trait definitions built when the `singleprover_circuit`
 //! argument is given to the macro
 
-use proc_macro2::TokenStream as TokenStream2;
+use proc_macro2::{Ident, TokenStream as TokenStream2};
```
Comment (akirillo) [inline circuit-macros/src/circuit_type/singleprover_circuit_types.rs:4]: getting an unused import warning for `Ident` here

File: circuit-types/src/fixed_point.rs [hunk 2]
```diff
@@ -47,14 +35,24 @@ lazy_static! {
...
-    };
+    pub static ref TWO_TO_NEG_M: ScalarField = TWO_TO_M_SCALAR.inverse().unwrap();
+}
+
+// -----------
+// | Helpers |
+// -----------
+
+/// Shift a given Scalar by M bits to the right
+pub fn shift_scalar_by_m(scalar: Scalar) -> Scalar {
```
Comment (akirillo) [inline circuit-types/src/fixed_point.rs:50]: nit: could call this `right_shift_scalar_by_m`, to make the direction clear from the fn name?

File: circuit-types/src/fixed_point.rs [hunk 3]
```diff
@@ -259,187 +276,99 @@ impl<'de> Deserialize<'de> for FixedPoint {
...
-impl<L: LinearCombinationLike> FixedPointVar<L> {
+impl FixedPointVar {
     /// Evaluate the given fixed point variable in the constraint system and
     /// return the underlying value as a floating point
     ///
     /// Note: not optimized, used mostly for tests
-    pub fn eval<CS: RandomizableConstraintSystem>(&self, cs: &CS) -> f32 {
+    pub fn eval_f64(&self, circuit: &PlonkCircuit) -> f64 {
         // Evaluate the scalar into a wide decimal form, shift it, then case
         // to f32
```
Comment (akirillo) [inline circuit-types/src/fixed_point.rs:292]:
```suggestion
        // Evaluate the scalar into a wide decimal form, shift it, then cast
        // to f64
```

**Review by akirillo — APPROVED** [review]
lgtm, couple nits

File: circuit-macros/src/circuit_type/singleprover_circuit_types.rs (hunk 1 above)
Comment (joeykraut) [reply circuit-macros/src/circuit_type/singleprover_circuit_types.rs:4]: resolved in a separate pr

**Pull request merged**